crates/model/src/quote.rs

```diff
@@ -230,6 +230,31 @@ impl Serialize for Validity {
     }
 }
 
+fn deserialize_optional_app_data<'de, D>(deserializer: D) -> Result<OrderCreationAppData, D::Error>
```

Comment:
This function seems to do the same thing as the derived `Deserialize` of `OrderCreationAppData` with the exception of a custom error message.
Is this really needed? And if it is would it make sense to remove the derived version and replace it with this logic so you don't have to use `deserialize_with = "deserialize_optional_app_data"`?

Reply:
This is needed because `#[serde(default, flatten)]` do not work together. Will add a comment.

Reply:
b0d90c1a175ed82a0233b0ba43560dc90bc5643e